### PR TITLE
Reduce GC pressure

### PR DESCRIPTION
**Pull request by tomusdrw**
## Summary by CodeRabbit

* **Refactor**
  * Execution now threads a shared runtime context and reuses pre-allocated result/argument objects for in-place updates.
  * Argument decoding uses instance-based filling with explicit low/high nibble helpers.
  * Memory APIs now use an explicit fault receiver to centralize page-fault reporting.

* **Bug Fixes**
  * Safer 32-bit signed-division edge-case handling.
  * More consistent branch/jump and memory outcome propagation.

* **Tests**
  * Tests updated to exercise the shared runtime context and fault-reporting behavior.

### CODE REVIEW

File: assembly/memory.ts [hunk 1]
```diff
@@ -164,237 +147,149 @@ export class Memory {
...
+        sourceIndex++;
+      }
       // move the pointers
-      sourceIndex += bytesToWrite;
       nextAddress += bytesToWrite;
     }
 
-    return NO_PAGE_FAULT;
+    return;
   }
```
[the remark this answers is not in the document]
Comment (tomusdrw) [reply assembly/memory.ts]: Already logged at #51
